README.md

```diff
@@ -1,31 +1,784 @@
+[![Version](https://img.shields.io/cocoapods/v/Fetcher.svg?style=flat)](http://cocoapods.org/pods/Fetcher)
+[![License](https://img.shields.io/cocoapods/l/Fetcher.svg?style=flat)](http://cocoapods.org/pods/Fetcher)
+[![Platform](https://img.shields.io/cocoapods/p/Fetcher.svg?style=flat)](http://cocoapods.org/pods/Fetcher)
+[![Slack Status](http://swiftkit.brightify.org//badge.svg)](http://swiftkit.brightify.org)
 
-## Usage
+## Introduction
+
+Fetcher is a small HTTP networking library for Swift. Its main goal is to simplify common tasks like sending REST requests. Networking is a very complex subject and our goal is not to cover everything that can be done. But we provide API that allows you to implement what you need or to customize behavior of Fetcher (this is handy if your server for some reason does not obey any standard).
+
+Main features:
+
+* Easy to use API
+* Compile time safety (as much as possible)
+* Abstraction of data encoding by using [DataMapper](https://github.com/Brightify/DataMapper)
+* Almost everything is customizable
+* Data mapping on background thread
+* Designed with DRY principle in mind
+* Support for RxSwift
 
-To run the example project, clone the repo, and run `pod install` from the Example directory first.
+## Changelog
 
-## [Documentation](https://github.com/brightify/SwiftKit/wiki)
+List of all changes and new features can be found [here](CHANGELOG.md).
 
 ## Requirements
 
+- **Swift 3**
+- **iOS 8+**
+
 ## Installation
 
-SwiftKit is available through [CocoaPods](http://cocoapods.org). To install
-it, simply add the following line to your Podfile:
+### CocoaPods
+
+Fetcher is available through [CocoaPods](http://cocoapods.org). To install
+it, simply add the following line to your test target in your Podfile:
+
+```ruby
+pod "Fetcher"
+```
+
+This will automatically include "Core" and "AlamofireRequestPerformer" subspecs.
+
+If you want only Fetcher without RequestPerformers (you have your own implementation), use:
+
+```ruby
+pod "Fetcher/Core"
+```
+
+Fetcher comes with support for [RxSwift](https://github.com/ReactiveX/RxSwift):
+
+```ruby
+pod "Fetcher/RxFetcher"
+```
+
+Note: "RxFetcher" subspec does not depend on "AlamofireRequestPerformer". So if you want to use it, you need to add it extra by:
+
+```ruby
+pod "Fetcher"
+```
+
+or: 
 
 ```ruby
-pod "SwiftKit"
+pod "Fetcher/AlamofireRequestPerformer"
+```
+
+## Usage
+
+Below is a complete list of all features this library offers and how to use them. Some examples of usage can be found in [tests](Tests). This documentation presumes that you are already familiar with [DataMapper](https://github.com/Brightify/DataMapper). Also you should know some basics about HTTP.
+
+### Quick overview
+
+Below is shown a simple use case. It does not explain all of the concepts used. Refer to corresponding chapters for explanations.
+
+Lets say we want to send `GET` request which will retrieve some data from your server. Our data looks like this:
```

Comment:
We refer to "your server" here, but in the next paragraph it becomes "our". I think it would feel much more friendly if we used "from our server".